my_project_omar/taskmanager.py

```diff
@@ -0,0 +1,47 @@
+from typing import Dict, Optional
+from modules import Task, UrgentTask, TaskCreate
+from uuid import UUID, uuid4
+
+tasks_dict = {}
+class TaskManager:
+    def create_task(self, task_data: TaskCreate) -> Task:
+        title=task_data.title
+        description=task_data.description
+        due_date=task_data.due_date
+        status=task_data.status
+          
+        if task_data.priority:
+            add_urgent=UrgentTask(task_data.priority,title,description,due_date,status)
+            tasks_dict[add_urgent.task_id]=add_urgent
+        else:
+            add_task=Task(title,description,due_date,status)
+            tasks_dict[add_task.task_id]=add_task
+        
+    def get_tasksbyid(self, task_id: str) -> Optional[Task]:
+      return self.tasks_dict.get(task_id)
+    
+    def get_tasks(self)-> Dict[str, Task]:
+      return self.tasks_dict
+#   Default value
+    def update_task(self, task_id: str, task_data: TaskCreate) -> Optional[Task]:
```

Comment:
You edited `update_task` in the `Tasks.py`, you should edit it here too.
Make sure that the user can edit one field at least without being forced to enter all fields. for example: the user can enter one value to edit and must not enter other values, so you edit the value is not none.